drivers/l3gxxxx/Kconfig

```diff
@@ -0,0 +1,432 @@
+    help
+        The typical sensitivity is 70 mdps.
+
+endif # !MODULE_A3G4250D
+
+endchoice
+
+choice
+    bool "Select filters used for output data"
+    default L3GXXXX_HPF_AND_LPF2
+    help
+        L3Gxxxx sensors integrate a combination of two low pass filters (LPF)
+        and one high pass filter (HPF) which can be used to filter the output
+        data. First, raw sensor data are always filtered by LPF1 with a cutoff
+        frequency that is fixed for the selected output data rate (ODR).
+        The resulting data can then optionally be filtered by HPF and LPF2.
+        Both filters can be used or bypassed.
+        Please note: The filter selection for the output data also affects the
+        filter selection for event interrupt generation. If the HPF is enabled
+        for filtering the output data, it is also active for filtering the
+        sensor data used for interrupt generation if the LPF2 is enabled for
+        interrupt generation.
+
+config L3GXXXX_NO_FILTER
+    bool "HPF and LPF2 are not used"
+
+config L3GXXXX_HPF_ONLY
+    bool "only HPF is used"
+
+config L3GXXXX_LPF2_ONLY
+    bool "only LPF2 used"
+
+config L3GXXXX_HPF_AND_LPF2
+    bool "HPF and LPF2 are used"
+
+endchoice
+
+if L3GXXXX_HPF_ONLY || L3GXXXX_HPF_AND_LPF2 || L3GXXXX_INT1_HPF_ONLY || L3GXXXX_INT1_HPF_AND_LPF2
+
+choice
+    bool "Select HPF X mode"
+    help
+        The high pass filter (HPF) can be used in different modes.
+
+config L3GXXXX_HPF_NORMAL
+    bool "Normal mode"
+    help
+        In Normal mode, the HPF is reset by reading REFERENCE register.
+
+config L3GXXXX_HPF_REFERENCE
+    bool "Reference mode"
+    help
+        In Reference mode, the output data are the difference to the
+        value written to the REFERENCE register.
+
+config L3GXXXX_HPF_AUTORESET
+    bool "Autoreset mode"
+    help
+        In Autoreset mode, the HPF is automatically reset when a configured
+        interrupt event occurs.
+
+endchoice
+
+config L3GXXXX_HPF_CUTOFF
+    int "Select HPF cutoff frequency n = [0...9]"
+    range 0 9
+    default 0
+    help
+        The cutoff frequency is specified as a value from 0 to 9, which defines
+        the cutoff frequency depending on the selected sensor mode and the
+        output data rate. Details can be found in the data sheet for the CTRL2
+        register field HPCF.
+
+endif # L3GXXXX_HPF_ONLY || L3GXXXX_HPF_AND_LPF2 || L3GXXXX_INT1_HPF_ONLY || L3GXXXX_INT1_HPF_AND_LPF2
+
+config MODULE_L3GXXXX_FIFO
+    bool "Support for the 32 level FIFO"
+    depends on MODULE_L3GXXXX
+    help
+        The 32 level FIFO is used to collect a number of sensor
+        data samples before the data have to be fetched by the MCU.
+
+if MODULE_L3GXXXX_FIFO
+
+choice
+    bool "Select FIFO mode"
+    default L3GXXXX_FIFO_MODE_FIFO
+    help
+        The 32 level FIFO is used to collect a number of sensor
+        data samples before the data have to be fetched by the MCU.
+
+config L3GXXXX_FIFO_MODE_BYPASS
```

Comment:
Why is there an option to have the FIFO but not use it?

Reply:
Good question. On the one hand it's for completeness to be compatible with the enumeration as defined in the datasheet. On the other hand, an application could change the FIFO mode during runtime, for example configure `BYPASS` mode first by default and switch to `STREAM` mode only after a configured event occured. The L3GD20H can do this automatically (`BYPASS_TO_STREAM`), for other variants of the sensor this has to be done by the application.

Reply:
Hmm I see. The on/off switch for the FIFO feature in the driver (`MODULE_L3GXXXX_FIFO`) makes totally sense, as it is a configuration of the driver itself. The choice of the initial mode is a bit more on the edge of being a hardware boot option IMO: it applies to a specific device instance, not to the driver itself. Still, we have been including these default values in Kconfig in many cases, so I'm fine with keeping this if it can be changed in runtime be the application.